src/Core/GraphQl/Type/FieldsBuilder.php

```diff
@@ -495,12 +519,20 @@ private function convertType(Type $type, bool $input, ?string $queryName, ?strin
         }
 
         if ($this->typeBuilder->isCollection($type)) {
-            $operationName = $queryName ?? $mutationName ?? $subscriptionName;
-
             return $this->pagination->isGraphQlEnabled($resourceClass, $operationName) && !$input ? $this->typeBuilder->getResourcePaginatedCollectionType($graphqlType, $resourceClass, $operationName) : GraphQLType::listOf($graphqlType);
         }
 
-        return $forceNullable || !$graphqlType instanceof NullableType || $type->isNullable() || (null !== $mutationName && 'update' === $mutationName)
+        $operation = null;
+        $resourceClass = '' === $resourceClass ? $rootResource : $resourceClass;
+        if ($resourceClass) {
+            $resourceMetadataCollection = $this->resourceMetadataCollectionFactory->create($resourceClass);
+            try {
+                $operation = $resourceMetadataCollection->getGraphQlOperation($operationName);
+            } catch (OperationNotFoundException $e) {
+            }
+        }
+
+        return $forceNullable || !$graphqlType instanceof NullableType || $type->isNullable() || ($operation instanceof Mutation || $operation instanceof Subscription)
```

Comment:
This is not the same as before. The type is nullable only when it's an `update` mutation.